src/kinetics/KineticsFactory.cpp

```diff
@@ -168,17 +171,44 @@ void addReactions(Kinetics& kin, const AnyMap& phaseNode, const AnyMap& rootNode
             AnyMap reactions = AnyMap::fromYamlFile(fileName,
                 rootNode.getString("__file__", ""));
             for (const auto& R : reactions[node].asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try{
+                    kin.addReaction(newReaction(R, kin));
+                }
+                catch(CanteraError& err){
+                    format_to(err_Plog_reactions, "{}", err.what());
+                }
             }
         } else {
             // specified section is in the current file
             for (const auto& R : rootNode.at(sections[i]).asVector<AnyMap>()) {
-                kin.addReaction(newReaction(R, kin));
+                try {
+                    kin.addReaction(newReaction(R, kin));
+                }
+                catch (CanteraError& err){
+                    std::string captured_err = err.what();
+                    captured_err.erase(std::remove(captured_err.begin(), captured_err.end(), '*'), captured_err.end());
+                    if (captured_err.find("Plog::validate") != std::string::npos) {
+                        captured_err.erase(0, 40);
+                        format_to(err_Plog_reactions, "{}", captured_err);
+                    } else {
+                        std::regex re("(\\w+)::(\\w+)");
+                        std::smatch match;
+                        regex_search(captured_err, match, re);
+                        std::string procedure = match[0];
+                        std::string msg = captured_err.substr(25+procedure.length(), -1);
+                        throw CanteraError(procedure,
+                            "{}", msg);
+                    }
+                }
             }
         }
     }
-
     kin.checkDuplicates();
+    if (!(to_string(err_Plog_reactions).empty())){
```

Comment:
This can be simplified to (and likewise in the other similar check):
```suggestion
    if (err_Plog_reactions.size()){
```